dll/win32/kernel32/winnls/string/locale.c

```diff
@@ -1445,463 +2003,1290 @@ BOOL WINAPI SetLocaleInfoW( LCID lcid, LCTYPE lctype, LPCWSTR data )
+ *            and dstlen != 0, and ERROR_INVALID_PARAMETER, if an invalid
+ *            parameter was given.
+ */
+INT WINAPI WideCharToMultiByte( UINT page, DWORD flags, LPCWSTR src, INT srclen,
+                                LPSTR dst, INT dstlen, LPCSTR defchar, BOOL *used )
+{
+    const union cptable *table;
+    int ret, used_tmp;
+
+    if (!src || !srclen || (!dst && dstlen) || dstlen < 0)
+    {
+        SetLastError( ERROR_INVALID_PARAMETER );
+        return 0;
+    }
+
+    if (srclen < 0) srclen = strlenW(src) + 1;
+
+    switch(page)
+    {
+    case CP_SYMBOL:
+        /* when using CP_SYMBOL, ERROR_INVALID_FLAGS takes precedence */
+        if (flags)
+        {
+            SetLastError( ERROR_INVALID_FLAGS );
+            return 0;
+        }
+        if (defchar || used)
+        {
+            SetLastError( ERROR_INVALID_PARAMETER );
+            return 0;
+        }
+        ret = wine_cpsymbol_wcstombs( src, srclen, dst, dstlen );
+        break;
+    case CP_UTF7:
+        /* when using CP_UTF7, ERROR_INVALID_PARAMETER takes precedence */
+        if (defchar || used)
+        {
+            SetLastError( ERROR_INVALID_PARAMETER );
+            return 0;
+        }
+        if (flags)
+        {
+            SetLastError( ERROR_INVALID_FLAGS );
+            return 0;
+        }
+        ret = utf7_wcstombs( src, srclen, dst, dstlen );
+        break;
+    case CP_UNIXCP:
+        if (unix_cptable)
+        {
+            ret = wine_cp_wcstombs( unix_cptable, flags, src, srclen, dst, dstlen,
+                                    defchar, used ? &used_tmp : NULL );
+            break;
+        }
+        /* fall through */
+    case CP_UTF8:
+        if (defchar || used)
+        {
+            SetLastError( ERROR_INVALID_PARAMETER );
+            return 0;
+        }
+        if (flags & ~WC_FLAGSMASK)
+        {
+            SetLastError( ERROR_INVALID_FLAGS );
+            return 0;
+        }
+        ret = wine_utf8_wcstombs( flags, src, srclen, dst, dstlen );
+        break;
+    default:
+        if (!(table = get_codepage_table( page )))
+        {
+            SetLastError( ERROR_INVALID_PARAMETER );
+            return 0;
+        }
+        if (flags & ~WC_FLAGSMASK)
+        {
+            SetLastError( ERROR_INVALID_FLAGS );
+            return 0;
+        }
+        ret = wine_cp_wcstombs( table, flags, src, srclen, dst, dstlen,
+                                defchar, used ? &used_tmp : NULL );
+        if (used) *used = used_tmp;
+        break;
+    }
+
+    if (ret < 0)
+    {
+        switch(ret)
+        {
+        case -1: SetLastError( ERROR_INSUFFICIENT_BUFFER ); break;
+        case -2: SetLastError( ERROR_NO_UNICODE_TRANSLATION ); break;
+        }
+        ret = 0;
+    }
+    TRACE("cp %d %s -> %s, ret = %d\n",
+          page, debugstr_wn(src, srclen), debugstr_an(dst, ret), ret);
+    return ret;
+}
+#endif // !__REACTOS__
```

Comment:
```suggestion
#ifdef __REACTOS__
#endif // WINVER >= 0x0600
#endif // __REACTOS__
```

Reply:
I improved the #ifdefs. I do not intend to add any new NT6 code in this PR.

Reply:
OK.